### DISCUSSION TITLE
Same model + different GPUs =  different transcriptions?

### DISCUSSION THREAD

**Discussion by joao** [openai/whisper#274] [Q&A] [answered]
Hi,

Was testing the `large` model on different GPUs to benchmark performance and noticed that the transcribed text was also different on some words.

Is this normal?
Shouldn't only time performance be affected by GPU characteristics? At least if both can fit the model in GPU RAM.

This was beetween a Nvidia T4 GPU 16GB and a Nvidia RTX 3060 12GB.

**Comment by joao** [comment, marked as answer]
Researching found the issue was between in one case executing via own Python code and via command line, not the different GPUs, that come with different defaults.

**Reply by jongwook** [reply]
Please note the command line uses heuristically chosen default settings for beam search and temperature-based fallback, etc.

**Reply by joao** [reply]
Managed to reproduce and fix the discrepancy I was having by following your answer here: https://github.com/openai/whisper/discussions/177#discussioncomment-3766539  
Thank you!

**Comment by jongwook** [comment]
Apart from slight numerical differences between hardware/drivers, a major source of nondeterminism comes from sampling using a nonzero temperature.

**Comment by Curtisfoster** [comment]
I am new and I have tried but I am unable to use it.can anyone help
me please?
